File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        unordered_map<char,int> map;
        string ans = "";
        
        for(auto& c : s)
        {
            map[c]++;
        }
        
        for(auto& c: order)
        {
            if(map.find(c) != map.end())
            {
                int count = map[c];
                
                
                while(count--)
                {
                    ans += c;
                    map[c]--;
                }
                    
                
            }
                
        }
        
        for(int c = 'a'; c <= 'z'; c++)
        {
            if(map[c] > 0)
            {
                while(map[c]--)
                {
                    ans += c;
                }
            }
        }
        
        return ans;
    }
};

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        hashmap = {}
        
        for c in s:
            hashmap[c] = hashmap.get(c, 0) + 1
        
        
        ans = ""
        for c in order:
            if c in hashmap:
                ans += c * hashmap[c]
                hashmap.pop(c)
                
        for c in hashmap:
            ans += c * hashmap[c]
        
        return ans